sormas-app/app/src/main/java/de/symeda/sormas/app/component/controls/ControlCheckBoxGroupFieldForMap.java

```diff
@@ -0,0 +1,205 @@
+				addItem(i, itemTotal - 1, items.get(i));
+			}
+
+			enumClass = c;
+			suppressListeners = false;
+		}
+	}
+
+	public void setReadOnly(boolean isReadOnly) {
+		int textColor = getResources().getColor(VisualState.NORMAL.getTextColor());
+		if (isReadOnly) {
+			checkBoxes.forEach((t, checkBox) -> {
+				checkBox.setEnabled(false);
+				checkBox.setTextColor(textColor);
+				checkBox.setButtonTintList(ColorStateList.valueOf(textColor));
+			});
+		}
+	}
+
+	private void addItem(int index, int lastIndex, Item<T> item) {
+		final CheckBox checkBox = createCheckBox(index, lastIndex, item);
+		checkBoxesFrame.addView(checkBox);
+		checkBoxes.put(item.getValue(), checkBox);
+	}
+
+	private CheckBox createCheckBox(int index, int lastIndex, Item<T> item) {
+		CheckBox checkBox = new CheckBox(new ContextThemeWrapper(getContext(), R.style.ControlCheckboxStyle));
+		checkBox.setId(index);
+		checkBox.setText(item.getKey());
+		checkBox.setTextSize(TypedValue.COMPLEX_UNIT_SP, 16);
+		checkBox.setOnCheckedChangeListener((b, i) -> {
+			onValueChanged();
+			if (inverseBindingListener != null) {
+				inverseBindingListener.onChange();
+			}
+		});
+		return checkBox;
+	}
+
+	private void uncheckAll() {
+		for (CheckBox checkBox : checkBoxes.values()) {
+			checkBox.setChecked(false);
+		}
+	}
+
+	private void removeAllItems() {
+		checkBoxes.clear();
+		checkBoxesFrame.removeAllViews();
+	}
+
+	@Override
+	protected void initialize(Context context, AttributeSet attrs, int defStyle) {
+		// Nothing to initialize
+	}
+
+	@Override
+	protected void inflateView(Context context, AttributeSet attrs, int defStyle) {
+		LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+
+		if (inflater != null) {
+			inflater.inflate(R.layout.control_checkboxgroup_layout, this);
+		} else {
+			throw new RuntimeException("Unable to inflate layout in " + getClass().getName());
+		}
+	}
+
+	@Override
+	protected void onFinishInflate() {
+		super.onFinishInflate();
+
+		checkBoxesFrame = this.findViewById(R.id.checkboxes_frame);
+	}
+
+	@Override
+	protected Map<T, Boolean> getFieldValue() {
+		Map<T, Boolean> selectedElements = new HashMap<>();
+
+		for (T key : checkBoxes.keySet()) {
+			if (checkBoxes.get(key).isChecked()) {
+				selectedElements.put(key, true);
+			}
+		}
+
+		return selectedElements;
+	}
+
+	@Override
+	protected void setFieldValue(Map<T, Boolean> value) {
+		if (value == null) {
+			uncheckAll();
+		} else {
+			for (Map.Entry<T, Boolean> element : value.entrySet()) {
+				CheckBox checkBox = checkBoxes.get(element.getKey());
+
+				if (checkBox == null) {
+					throw new IllegalArgumentException(
+						"Passed list arguments contains an element that is not part of this ControlCheckBoxGroupField");
+				}
+
+				checkBox.setChecked(element.getValue());
+			}
+		}
+	}
+
+	@Override
+	public void setEnabled(boolean enabled) {
+		// TODO: Implement; Probably replace elements with tags
```

Comment:
this logic from `setReadonly` should go here and there will be no need for that method

Reply:
moved the code to setEnabled method